### PR TITLE
Update ecdh.h

### PR DESCRIPTION
**Pull request by RosZad**
Minor documentation improvements:
*Standardized file brief description.
*Separated return statements.
*Reordered tags within documentation blocks so that params and returns are last in block.
*p_rng descriptions changed from "parameter" to "context".
*Removed bullets from parameter descriptions.

### CODE REVIEW

File: include/mbedtls/ecdh.h [hunk 1]
```diff
@@ -40,14 +41,12 @@ extern "C" {
...
- * Defines the source of the imported EC key:
- * <ul><li>Our key.</li>
- * <li>The key of the peer.</li></ul>
+ * Defines the source of the imported EC key.
  */
 typedef enum
 {
-    MBEDTLS_ECDH_OURS,
-    MBEDTLS_ECDH_THEIRS,
+    MBEDTLS_ECDH_OURS, /**< Our key. */
```
Comment (hanno-arm) [inline include/mbedtls/ecdh.h]: Minor style issue: Please align the two comments.

File: include/mbedtls/ecdh.h [hunk 2]
```diff
@@ -97,21 +98,22 @@ int mbedtls_ecdh_gen_public( mbedtls_ecp_group *grp, mbedtls_mpi *d, mbedtls_ecp
  *                  implemented during the ECDH key exchange. The first core
  *                  computation is performed by mbedtls_ecdh_gen_public().
  *
+ * \see             ecp.h
+ *
+ * \note            If \p f_rng is not NULL, it is used to implement
+ *                  countermeasures against potential elaborate timing
```
Comment (hanno-arm) [inline include/mbedtls/ecdh.h]: Suggestion: Don't restrict to timing attacks, but instead say `... it is used to implement countermeasures against side-channel attacks. For more information, ...`. @yanesca Ping.

File: include/mbedtls/ecdh.h [hunk 3]
```diff
@@ -249,21 +252,21 @@ int mbedtls_ecdh_read_public( mbedtls_ecdh_context *ctx,
  *                  This is the last function used by both TLS client
  *                  and servers.
  *
+ * \note            If \p f_rng is not NULL, it is used to implement
+ *                  countermeasures against potential elaborate timing
```
Comment (hanno-arm) [inline include/mbedtls/ecdh.h]: Same comment as above here.

File: include/mbedtls/ecdh.h (hunk 1 above)
Comment (AndrzejKurek) [reply include/mbedtls/ecdh.h]: Again, I find these comments superfluous.

Comment (RosZad) [reply include/mbedtls/ecdh.h]: aligned.

File: include/mbedtls/ecdh.h (hunk 2 above)
Comment (RosZad) [reply include/mbedtls/ecdh.h]: done

File: include/mbedtls/ecdh.h (hunk 3 above)
Comment (RosZad) [reply include/mbedtls/ecdh.h]: done

File: include/mbedtls/ecdh.h [hunk 4]
```diff
@@ -1,10 +1,11 @@
 /**
  * \file ecdh.h
  *
- * \brief The Elliptic Curve Diffie-Hellman (ECDH) protocol APIs.
- *
- * ECDH is an anonymous key agreement protocol allowing two parties to
- * establish a shared secret over an insecure channel. Each party must have an
+ * \brief This file contains ECDH definitions and functions.
```
[the remark this answers is not in the document]
Comment (RosZad) [reply include/mbedtls/ecdh.h]: Again, the brief file description states what it contains. This information was not removed, but simply made part of the file's detailed description.

**Pull request merged**